src/features/auth/components/WrongChain.tsx

```diff
@@ -1,15 +1,31 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Button } from "components/ui/Button";
 
 import alert from "assets/icons/expression_alerted.png";
+import metamaskIcon from "assets/icons/metamask-icon.png";
+import { metamask } from "lib/blockchain/metamask";
 
 export const WrongChain: React.FC = () => {
+  const [isDefaultNetwork, setIsDefaultNetwork] = useState(false);
   const goToPolygonSetupDocs = () => {
     window.open(
       "https://docs.sunflower-land.com/guides/getting-setup#polygon-setup",
       "_blank"
     );
   };
+  
+  useEffect(() => {
+    const load = async () => {
+      let defaultNetworkStatus = await metamask.checkDefaultNetwork();
+      setIsDefaultNetwork(defaultNetworkStatus);
+    };
+
+    load();
+  });
```

Comment:
If you want this code to run only once on mount, I would suggest adding empty dependencies. Without these, this code will run on every re-render

i.e.
```
useEffect(() => {
    const load = async () => {
      let defaultNetworkStatus = await metamask.checkDefaultNetwork();
      setIsDefaultNetwork(defaultNetworkStatus);
    };

    load();
  }, []);
```